### PR TITLE
Optimize hetSNP file generation

### PR DESCRIPTION
**Pull request by zackgomez**
Batch calls to tabix to reduce process creation overhead.  This is at least 100x faster than the previous code.  The limiting factor is now the time it takes GffTranscriptReader to parse the reference files 

This requires tabix 1.12 and the ubuntu installed tabix is 1.10.  So I also include a dockerfile change to build tabix 1.14 from source in a separate stage.

### CODE REVIEW

**Comment by x811zou** [comment]
Optimization in code look great!
@tncowart can you review the docker changes?

File: BEASTIE/extractHets.py
Comment (tncowart) [inline BEASTIE/extractHets.py:68]: Can you delete the commented out code here?

File: BEASTIE/extractHets.py
Comment (tncowart) [inline BEASTIE/extractHets.py:87]: Instead of adding all the outputs to an array, then iterating through the array, I think it would be better to skip the `outputs` array entirely and process the results from `tabix` immediately. Maybe even pull the tabix output processing code into its own function.

File: BEASTIE/extractHets.py
Comment (tncowart) [inline BEASTIE/extractHets.py:98]: What is this for? I feel this assert depends on an interaction between lines 55 and line 113. is it supposed to be 
```suggestion
                    assert transcripts is not None
```

File: BEASTIE/extractHets.py
Comment (tncowart) [inline BEASTIE/extractHets.py:110]: I think any `continue`s (or other guard statements) should be moved up as high as possible.

File: BEASTIE/extractHets.py
Comment (tncowart) [inline BEASTIE/extractHets.py:102]: If it's a bad line this code will crash at line 108. I think you should handle that error here somehow (continue? exit?) rather than let that happen.

File: BEASTIE/extractHets.py
Comment (tncowart) [inline BEASTIE/extractHets.py:114]: pos is an int (see line 106), you don't need to cast here.

**Review by tncowart — CHANGES_REQUESTED** [review]
Looking good. I have a few change requests, but no major surgery or anything.

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:68]: This code is here as it helped expose a difference in functionality between the optimized and unoptimized versions.  @x811zou  has said that the difference is actually an improvement but for this initial diff I wanted to retain the exact same behavior while improving performance.

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:87]: So my thought here is to process this in 3 explicit stages

1. compute regions to collect
2. collect regions in batches
3. process regions

I agree that it is an inefficiency and could be written as

1. compute regions
2. collected and process regions in batch

I don't feel super strongly about this, but would prefer to leave it as-is as I know it works.

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:102]: I will replace with an assert

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:120]: One of the few comments I disagree with.

I think join is superior as a change to the separator would be easier to implement and it can be more clearly read as this multiline array format

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:42]: removed

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:20]: (y)

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:120]: I do think I could remove all the `str()` calls

Comment (zackgomez) [reply BEASTIE/extractHets.py:120]: Actually I cannot!

File: BEASTIE/extractHets.py
Comment (zackgomez) [reply BEASTIE/extractHets.py:101]: Moved up as high as could be

File: BEASTIE/extractHets.py
Comment (tncowart) [reply BEASTIE/extractHets.py:114]: Well, changing the separator in either case is trivial. But if you like join more, I'm not going to force the issue.

File: BEASTIE/extractHets.py
Comment (tncowart) [reply BEASTIE/extractHets.py:65]: Okay, that's fine. Can you write that in a comment in the code? I think it's good practice to explain why commented-out code hasn't just been deleted alongside said code.

**Review by tncowart — APPROVED** [review]
I'd prefer you add a comment to the commented code, but I don't think it needs to hold up approval. Looks good!

**Review by tncowart — APPROVED** [review]

**Pull request merged**
